src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessageData] = useState(chatMessages);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const updateMessageData = (mId) => {
+    const newChatMessages = [...messages];
+
+    for (let thisMessage of newChatMessages) {
+      if (thisMessage.id === mId) {
+        let newMessage = thisMessage;
+        if (!newMessage.liked) {
+          newMessage.liked = true;
+          setLikesCount(likesCount + 1);
+        } else if (newMessage.liked) {
+          newMessage.liked = false;
+          setLikesCount(likesCount - 1);
```

Comment:
So here's what I was talking about earlier. We end up running the setLikesCount function for each message that we iterate through. This just tends to be a bit more inefficient than if we created a single function that simply counted all the likes once the list has been updated. We could then call that function and make it the likesCount variable to be used in the display.